homework/src/index.js

```diff
@@ -30,17 +32,109 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { html: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { html: JSON.stringify(data, null, 2) });
+  async function main(url) {
+    const root = document.getElementById('root');
+    const header = createAndAppend('header', root);
+    createAndAppend('p', header, {html: 'HYF Repositories', class: 'header-lable'});
+    const select = createAndAppend('select', header, { id: 'selectMenu' });
+    // use promise object
+    try {
+      const data = await fetchJSON(url);
+        data.sort((a, b) => a.name.localeCompare(b.name, 'fr', {ignorePunctuation: true}));
+        data.forEach((repository, index) => {
+          createAndAppend('option', select, {html: repository.name, value: index});
+        });
+        const container = createAndAppend('div', root, { id: 'container'});
+        contentLeft(container, data, select);
+        contentRight(container, data, select);
+    }
+    catch(error) {
+      createAndAppend('div', root, { html: error.message, class: 'alert-error' });
+    }
+  }
+  
+  // function who creates a left content section
+  function contentLeft(container, data, select) {
+    const leftSection = createAndAppend('div', container, { id: 'leftSection', class: 'whiteframe' });
+    showTableRepo(leftSection, data[0]);
+
+    select.addEventListener('change', (event) => {
+      leftSection.innerHTML = "";
+      const theEventIndex = event.target.value;
+      showTableRepo(leftSection, data[theEventIndex]);
```

Comment:
Can't we just move the `data[theEventIndex]` to it's own variable, it's the selected repository, Right?